WordPress/Classes/ViewRelated/Reader/Manage/ReaderTagsTableViewController.swift

```diff
@@ -0,0 +1,166 @@
+class ReaderTagsTableViewController: UIViewController {
+
+    private let style: UITableView.Style
+
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero, style: style)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        return tableView
+    }()
+
+    private var tableViewHandler: OffsetTableViewHandler?
+
+    init(style: UITableView.Style) {
+        self.style = style
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(tableView)
+        view.pinSubviewToAllEdges(tableView)
+
+        setupTableHandler()
+    }
+
+    private func setupTableHandler() {
+        let handler = OffsetTableViewHandler(tableView: tableView)
+        handler.delegate = self
+        tableViewHandler = handler
+    }
+}
+
+// MARK: - WPTableViewHandlerDelegate
+
+extension ReaderTagsTableViewController: WPTableViewHandlerDelegate {
+
+    func managedObjectContext() -> NSManagedObjectContext {
+        return ContextManager.sharedInstance().mainContext
+    }
+
+    func fetchRequest() -> NSFetchRequest<NSFetchRequestResult> {
+        return ReaderTagTopic.tagsFetchRequest
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard tableViewHandler?.adjusted(indexPath: indexPath) == nil else { return }
+        tableView.deselectSelectedRowWithAnimation(true)
+        showAddTag()
+    }
+
+    func configureCell(_ cell: UITableViewCell, at indexPath: IndexPath) {
+        let topic = tableViewHandler?.object(at: indexPath) as? ReaderTagTopic
+        configure(cell: cell, for: topic)
+    }
+}
+
+// MARK: - Actions
+
+extension ReaderTagsTableViewController {
+    @objc func tappedAccessory(_ sender: UIButton) {
+        guard let point = sender.superview?.convert(sender.center, to: tableView),
+            let indexPath = tableView.indexPathForRow(at: point),
+            let adjustIndexPath = tableViewHandler?.adjusted(indexPath: indexPath),
+            let topic = tableViewHandler?.resultsController.object(at: adjustIndexPath) as? ReaderTagTopic else { return }
```

Comment:
The `return` should be on a new line. ([style guide](https://github.com/wordpress-mobile/swift-style-guide))